src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.cs

```diff
@@ -255,7 +263,7 @@ private static SyntaxNode GetContainer(BinaryExpressionSyntax isExpression)
                 {
                     if (child.IsNode)
                     {
-                        AddMatches(child.AsNode(), expr, type, matches);
+                        AddMatches(child.AsNode()!, expr, type, matches);
```

Comment:
Unrelated to the PR: I believe there is a way for `AsNode` to be annotated that it's not null if `IsNode` is true and eliminate the null-suppression. This will be a good to have given that `AsNode` is a public API.

Reply:
I don't believe that is true. Can you share how though you think that would work?  Thanks!

Reply:
I was thinking of `MemberNotNullWhen`, but yeah it doesn't work for methods.